lib/circuitbox/circuit_breaker.rb

```diff
@@ -112,8 +112,16 @@ def try_close_next_time
     end
 
   private
+
     def should_open?
-      passed_volume_threshold? && passed_rate_threshold?
```

Comment:
I appreciate the named methods here lending to the readability of this method.  Can we keep the readability while fixing the bug?

Reply:
We would have to change quite a bit about those methods, essentially they just become one liners that are only used in one place.

Something like this might work

```ruby
def should_open?
  failures = failure_count
  successes = success_count
  rate = error_rate(failures, successes)

  log_metrics(rate, failures, successes)

  passed_volume_threshold?(failures, successes) && passed_rate_threshold?(rate)
end

def passed_volume_threshold?(failures, successes)
  failures + successes > option_value(:volume_threshold)
end

def passed_rate_threshold?(rate)
  rate >= option_value(:error_threshold)
end
```

I'm not sure how much this increases readability though